Libraries/LibWeb/CSS/Interpolation.cpp

```diff
@@ -577,6 +577,14 @@ NonnullRefPtr<CSSStyleValue const> interpolate_value(DOM::Element& element, Calc
         return delta >= 0.5f ? to : from;
     }
 
+    static auto interpolate_length_percentage = [](LengthPercentage const& from, LengthPercentage const& to, float delta) -> Optional<LengthPercentage> {
+        if (from.is_length() && to.is_length())
+            return Length::make_px(interpolate_raw(from.length().raw_value(), to.length().raw_value(), delta));
+        if (from.is_percentage() && to.is_percentage())
+            return Percentage(interpolate_raw(from.percentage().value(), to.percentage().value(), delta));
+        return {};
```

Comment:
The `if (from.type() != to.type()) {)` block at the top of `interpolate_value()` handles interpolating percentage/dimension values. You might be able to use that here somehow, but no rush to do it right now.